### PR TITLE
Update to the new crypto API

### LINKED ISSUE #18: Prepare for OTP 24
`:crypto.hmac` is deprecated and will be removed in OTP 24. See https://github.com/hexpm/hex/issues/844 for more information.

### PR DESCRIPTION
**Pull request by ConnorRigby**
Fixes #18

### CODE REVIEW

File: lib/pbcs/aes_cbc_hmac_sha2.ex [hunk 1]
```diff
@@ -3,6 +3,20 @@ defmodule PBCS.AES_CBC_HMAC_SHA2 do
 
   @behaviour ContentEncryptor
 
+  defmodule CryptoWrapper do
```
Comment (fhunleth) [inline lib/pbcs/aes_cbc_hmac_sha2.ex:6]:
```suggestion
  defmodule CryptoWrapper do
    @moduledoc false
```

**Review by fhunleth — APPROVED** [review]

**Comment by mobileoverlord** [comment]
All tests fail when testing on OTP 24 with elixir 1.12.0-rc.1 for OTP 24.

```
  1) test python passlib pbkdf2_sha512 (PBCSTest)
     test/pbcs_test.exs:87
     ** (UndefinedFunctionError) function :crypto.hmac/3 is undefined or private, use crypto:mac/4 instead
     code: |> check_vectors(19_000, 64, :sha512)
     stacktrace:
       (crypto 5.0) :crypto.hmac(:sha512, "", "")
       (pbcs 0.1.2) lib/pbcs/pkcs5.ex:7: PBCS.PKCS5.pbkdf2/5
       test/pbcs_test.exs:13: anonymous fn/4 in PBCSTest.check_vectors/4
       (elixir 1.12.0-rc.1) lib/enum.ex:930: Enum."-each/2-lists^foreach/1-0-"/2
       test/pbcs_test.exs:112: (test)



  2) test base pbkdf2_sha512 (PBCSTest)
     test/pbcs_test.exs:65
     ** (UndefinedFunctionError) function :crypto.hmac/3 is undefined or private, use crypto:mac/4 instead
     code: |> check_vectors(100_000, 64, :sha512)
     stacktrace:
       (crypto 5.0) :crypto.hmac(:sha512, "", "")
       (pbcs 0.1.2) lib/pbcs/pkcs5.ex:7: PBCS.PKCS5.pbkdf2/5
       test/pbcs_test.exs:13: anonymous fn/4 in PBCSTest.check_vectors/4
       (elixir 1.12.0-rc.1) lib/enum.ex:930: Enum."-each/2-lists^foreach/1-0-"/2
       test/pbcs_test.exs:84: (test)



  3) test base (PBCSTest)
     test/pbcs_test.exs:55
     ** (UndefinedFunctionError) function :crypto.hmac/3 is undefined or private, use crypto:mac/4 instead
     code: assert pbkdf2("password", "0123456789abcdef", 1, 32, :sha512) ==
     stacktrace:
       (crypto 5.0) :crypto.hmac(:sha512, "", "")
       (pbcs 0.1.2) lib/pbcs/pkcs5.ex:7: PBCS.PKCS5.pbkdf2/5
       test/pbcs_test.exs:56: (test)



  4) test PKCS#5 test vectors (PBCSTest)
     test/pbcs_test.exs:17
     ** (UndefinedFunctionError) function :crypto.hmac/3 is undefined or private, use crypto:mac/4 instead
     code: |> check_vectors()
     stacktrace:
       (crypto 5.0) :crypto.hmac(:sha, "", "")
       (pbcs 0.1.2) lib/pbcs/pkcs5.ex:7: PBCS.PKCS5.pbkdf2/5
       test/pbcs_test.exs:7: anonymous fn/1 in PBCSTest.check_vectors/1
       (elixir 1.12.0-rc.1) lib/enum.ex:930: Enum."-each/2-lists^foreach/1-0-"/2
       test/pbcs_test.exs:52: (test)



Finished in 0.02 seconds (0.02s async, 0.00s sync)
4 tests, 4 failures
```

**Comment by mobileoverlord** [comment]
Can you fix the test suite so it runs for OTP 24 as well as previous versions? I'll work on migrating / updating CI to GitHub actions.

**Comment by ConnorRigby** [comment]
Sorry about that. i missed a couple things. I've updated the PR and the tests pass locally on the official otp 24 release

**Review by mobileoverlord — APPROVED** [review]

**Pull request merged**